agrirouter-sdk-dotnet-standard-impl/Service/Messaging/Abstraction/SendMessageBaseService.cs

```diff
@@ -86,6 +87,83 @@ public MessagingResult Send(SendMessageParameters sendMessageParameters)
+            }
+
+            if (MessageCanBeChunked(sendMessageParameters.TechnicalMessageType))
+            {
+                if (MessageHasToBeChunked(sendMessageParameters.Base64MessageContent))
+                {
+                    var chunkContextId = Guid.NewGuid().ToString();
+                    var totalSize = Encoding.Unicode.GetByteCount(sendMessageParameters.Base64MessageContent);
+
+                    var chunkedMessages = ChunkMessageContent(sendMessageParameters.Base64MessageContent,
+                        sendMessageParameters.ChunkSize > 0
+                            ? sendMessageParameters.ChunkSize
+                            : ChunkSizeDefinition.MaximumSupported);
+
+                    var current = 0;
+                    foreach (var chunkedMessage in chunkedMessages)
+                    {
+                        var sendMessageParametersDuplicate = new SendChunkedMessageParameters
+                        {
+                            Recipients = sendMessageParameters.Recipients,
+                            TypeUrl = sendMessageParameters.TypeUrl,
+                            TechnicalMessageType = sendMessageParameters.TechnicalMessageType,
+                            ApplicationMessageId = MessageIdService.ApplicationMessageId()
+                        };
+                        var chunkComponent = new ChunkComponent
+                        {
+                            Current = current++,
+                            Total = chunkedMessage.Length,
+                            ContextId = chunkContextId,
+                            TotalSize = totalSize
+                        };
+                        sendMessageParametersDuplicate.ChunkInfo = chunkComponent;
+                        sendMessageParametersDuplicate.Base64MessageContent = chunkedMessage;
+                        encodedMessages.Add(Encode(sendMessageParametersDuplicate).Content);
+                    }
+                }
+                else
+                {
+                    encodedMessages = new List<string> {Encode(sendMessageParameters).Content};
+                }
+            }
+            else
+            {
+                encodedMessages = new List<string> {Encode(sendMessageParameters).Content};
+            }
+
+            var messagingParameters = sendMessageParameters.BuildMessagingParameter(encodedMessages);
+            return _messagingService.SendAsync(messagingParameters);
+        }
+
+        public Task<MessagingResult> SendAsync(SendProtobufMessageParameters sendMessageParameters)
+        {
+            List<string> encodedMessages;
+
+            if (sendMessageParameters.ProtobufMessageContent == null)
+            {
+                throw new CouldNotSendEmptyMessageException("Sending empty messages does not make any sense.");
+            }
+            
+            encodedMessages = new List<string> {Encode(sendMessageParameters).Content};
```

Comment:
Join declaration and assignment.

```suggestion
            if (sendMessageParameters.ProtobufMessageContent == null)
            {
                throw new CouldNotSendEmptyMessageException("Sending empty messages does not make any sense.");
            }
            
            var encodedMessages = new List<string> {Encode(sendMessageParameters).Content};
```

Reply:
Fixed